src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -755,6 +758,43 @@ namespace winrt::TerminalApp::implementation
         _ApplyTheme(_settings->GlobalSettings().GetTheme());
     }
 
+    fire_and_forget AppLogic::_ApplyStartupTaskStateChange()
+    {
+        co_await winrt::resume_foreground(_root->Dispatcher());
+
+        StartupTaskState state;
+        bool tryEnableStartupTask = _settings->GlobalSettings().StartOnUserLogin();
+        StartupTask task = co_await StartupTask::GetAsync(StartupTaskName);
+
+        state = task.State();
+        switch (state)
+        {
+        case StartupTaskState::Disabled:
+        {
+            if (tryEnableStartupTask)
+            {
+                co_await task.RequestEnableAsync();
+            }
+            break;
+        }
+        case StartupTaskState::DisabledByUser:
+        {
+            // TODO: how should we handle this?

```

Comment:
I feel like for this (and the other `StartupTaskState`) we should display a warning of some sort. But I'd be ok with just ignoring these cases for now and linking a new GitHub issue for a separate PR.

Reply:
Created #6254